lti/services/auth_data.py

```diff
@@ -36,60 +39,63 @@ class AuthDataService(object):
 
     """
 
-    def __init__(self, open_=None):
-        self.open_ = open_ or open  # Test seam.
-        self._name = 'canvas-auth.json'
-        self.auth_data = {}
-        self.load()
+    def __init__(self, db):
+        self._db = db
 
     def set_tokens(self, oauth_consumer_key, lti_token, lti_refresh_token):
-        assert oauth_consumer_key in self.auth_data
-        lock = filelock.FileLock("authdata.lock")
-        with lock.acquire(timeout=1):
-            self.auth_data[oauth_consumer_key]['lti_token'] = lti_token
-            self.auth_data[oauth_consumer_key]['lti_refresh_token'] = lti_refresh_token
-            self.save()
+        try:
+            credentials = self._credentials_for(oauth_consumer_key)
+        except KeyError:
+            # We raise AssertionError here just to maintain compatibility with
+            # the legacy API of auth_data, for now.
+            raise AssertionError
+
+        # Delete all existing access tokens for these credentials.
+        for access_token in credentials.access_tokens:
+            self._db.delete(access_token)
```

Comment:
When it was using the `canvas-auth.json()` file `set_tokens()` would _replace_ any existing access token and refresh token in the file for this consumer key with the new ones. In the DB we can actually have multiple access and refresh tokens for a single credentials, but for now we're maintaining the behaviour of using just one, so we have to delete existing ones before adding a new one.